tests/gateway.rs

```diff
@@ -0,0 +1,62 @@
+use k8s_gateway_api::v1alpha2::{
+    AllowedRoutes, Gateway, GatewaySpec, GatewayTlsConfig, Listener, RouteGroupKind,
+    SecretObjectReference,
+};
+use kube::{api::PostParams, core::ObjectMeta};
+
+#[tokio::test(flavor = "current_thread")]
+async fn round_trip() {
+    tracing::trace!("Initializing client");
+    let client = kube::Client::try_default()
+        .await
+        .expect("failed to initialize k8s client");
+    let api = kube::Api::<Gateway>::default_namespaced(client);
+
+    let gateway = Gateway {
+        metadata: ObjectMeta {
+            name: Some("prod-web".to_string()),
+            ..Default::default()
+        },
+        spec: GatewaySpec {
+            gateway_class_name: "acme-lb".to_string(),
+            listeners: vec![Listener {
+                protocol: "HTTPS".to_string(),
+                port: 443,
+                allowed_routes: Some(AllowedRoutes {
+                    namespaces: None,
+                    kinds: Some(vec![RouteGroupKind {
+                        group: Some("gateway.networking.k8s.io".to_string()),
+                        kind: "HTTPRoute".to_string(),
+                    }]),
+                }),
+                tls: Some(GatewayTlsConfig {
+                    mode: None,
+                    options: None,
+                    certificate_refs: Some(vec![SecretObjectReference {
+                        group: None,
+                        kind: None,
+                        name: "admin-controlled-cert".to_string(),
+                        namespace: None,
```

Comment:
tioli: here and elsewhere, you can probably use the [struct update syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax):

```suggestion
                        name: "admin-controlled-cert".to_string(),
                        ..SecretObjectReference::default()
```